foxbot_core.ino

```diff
@@ -221,6 +228,26 @@ void loop() {
 		odom.twist.twist.angular.z = angular_velocity_est;
 
     	odom_publisher.publish(&odom);
+		
+		// Calculate joint_state message
+		motor_left_rate_est= motor_left_rate_est*(-1);
```

Comment:
from the doc, the position of a revolute joint is an angle (the mechanical angle of the wheel wrt to the chassis), so here we should integrate the angular rate with:
left_wheel_position += motor_left_rate_est * dt;
right_wheel_position += right_left_rate_est * dt;